tests/components/aurora_abb_powerone/test_config_flow.py

```diff
@@ -1,165 +1,592 @@
         ) as mock_setup_entry,
     ):
-        result2 = await hass.config_entries.flow.async_configure(
-            result["flow_id"],
-            {CONF_PORT: "/dev/ttyUSB7", CONF_ADDRESS: 7},
+        create_entry_result = await hass.config_entries.flow.async_configure(
+            configure_tcp_result["flow_id"],
+            {
+                CONF_TCP_HOST: "127.0.0.1",
+                CONF_TCP_PORT: 502,
+                CONF_INVERTER_SERIAL_ADDRESS: 2,
+            },
         )
 
-    assert result2["type"] is FlowResultType.CREATE_ENTRY
-
-    assert result2["data"] == {
-        CONF_PORT: "/dev/ttyUSB7",
-        CONF_ADDRESS: 7,
-        ATTR_FIRMWARE: "1.234",
-        ATTR_MODEL: "9.8.7.6 (A.B.C)",
-        ATTR_SERIAL_NUMBER: "9876543",
-        "title": "PhotoVoltaic Inverters",
+    assert create_entry_result["type"] is FlowResultType.CREATE_ENTRY
+    assert create_entry_result["title"] == DEFAULT_INTEGRATION_TITLE
+    assert create_entry_result["data"] == {
+        CONF_TCP_HOST: "127.0.0.1",
+        CONF_TCP_PORT: 502,
+        CONF_INVERTER_SERIAL_ADDRESS: 2,
+        CONF_TRANSPORT: TRANSPORT_TCP,
+        ATTR_SERIAL_NUMBER: "TCP-123456",
+        ATTR_MODEL: "M1.2.3 (XYZ)",
+        ATTR_FIRMWARE: "2.0.0",
+        "title": DEFAULT_INTEGRATION_TITLE,
     }
     await hass.async_block_till_done()
-    assert len(mock_setup.mock_calls) == 1
     assert len(mock_setup_entry.mock_calls) == 1
 
 
-async def test_form_no_comports(hass: HomeAssistant) -> None:
-    """Test we display correct info when there are no com ports.."""
+async def test_tcp_error_cannot_connect(hass: HomeAssistant) -> None:
+    """Test TCP path error handling -> cannot_connect."""
+    choose_transport_result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert choose_transport_result["type"] is FlowResultType.FORM
+    assert choose_transport_result["step_id"] == "choose_transport"
+
+    configure_tcp_result = await hass.config_entries.flow.async_configure(
+        choose_transport_result["flow_id"], {CONF_TRANSPORT: TRANSPORT_TCP}
+    )
+    assert configure_tcp_result["type"] is FlowResultType.FORM
+    assert configure_tcp_result["step_id"] == "configure_tcp"
+
+    client = MagicMock()
+    client.try_connect_and_fetch_identifier.side_effect = AuroraClientError(
+        "some tcp error"
+    )
 
-    fakecomports = []
     with patch(
-        "serial.tools.list_ports.comports",
-        return_value=fakecomports,
+        "homeassistant.components.aurora_abb_powerone.config_flow.AuroraClient.from_tcp",
+        return_value=client,
     ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        form_result = await hass.config_entries.flow.async_configure(
+            configure_tcp_result["flow_id"],
+            {
+                CONF_TCP_HOST: "192.0.2.10",
```

Comment:
Any reason for picking a host that could be on the network?
Picking localhost can avoid a lengthy timeout in case something goes wrong with the test.
```suggestion
                CONF_TCP_HOST: "127.0.0.1",
```